test/unit/mdc-tabs/mdc-tab-bar.test.js

```diff
@@ -0,0 +1,328 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assert} from 'chai';
+import bel from 'bel';
+import domEvents from 'dom-events';
+import td from 'testdouble';
+import {MDCTabBar} from '../../../packages/mdc-tabs/tab-bar';
+import {MDCTabBarFoundation} from '../../../packages/mdc-tabs/tab-bar';
+import {createMockRaf} from '../helpers/raf';
+
+class MockTab {
+  constructor() {
+    this.root = bel`<a class="mdc-tab">Item</a>`;
+    this.measureSelf = td.func('measureSelf');
+    this.isActive = false;
+    this.preventDefaultOnClick = false;
+    this.computedWidth = 100;
+    this.computedLeft = 200;
+  }
+}
+
+function getFixture() {
+  return bel`
+    <div>
+      <nav class="mdc-tab-bar">
+        <a class="mdc-tab" href="#one">Item One</a>
+        <a class="mdc-tab" href="#two">Item Two</a>
+        <a class="mdc-tab" href="#three">Three</a>
+        <span class="mdc-tab-bar__indicator"></span>
+      </nav>
+    </div>`;
+}
+
+function setupTest() {
+  const MockTabBarFoundation = td.constructor(MDCTabBarFoundation);
+  const foundation = new MockTabBarFoundation();
+  const mockTab = new MockTab();
+  const fixture = getFixture();
+  const root = fixture.querySelector('.mdc-tab-bar');
+  const indicator = fixture.querySelector('.mdc-tab-bar__indicator');
+  const component = new MDCTabBar(root, undefined, () => mockTab);
+
+  return {fixture, root, indicator, component, foundation};
+}
+
+suite('MDCTabBar');
+
+test('attachTo returns a component instance', () => {
+  assert.isOk(MDCTabBar.attachTo(setupTest().root) instanceof MDCTabBar);
+});
+
+test('#get tabs returns tabs', () => {
+  const {component} = setupTest();
+
+  assert.isArray(component.tabs);
+
+  component.tabs.forEach((tab) => {
+    assert.instanceOf(tab, MockTab);
+  });
+});
+
+test('#get activeTab returns active tab', () => {
+  const {component} = setupTest();
+  const targetIndex = 1;
+  const tab = component.tabs[targetIndex];
+
+  tab.isActive = true;
+  assert.equal(tab, component.activeTab);
+});
+
+test('#set activeTab proxies to foundation_.switchToTabAtIndex', () => {
```

Comment:
(here and below) rather than saying "proxies to foundation_.", which describes an implementation detail not seen in this test, it would be better to describe what the implementation is trying to accomplish